core/modules/system/system.install

```diff
@@ -9,6 +9,7 @@
  */
 function system_requirements($phase) {
   global $base_url;
+  $site_config = config('system.core');
```

Comment:
It might make sense to move this down to the phase and condition where it's actually needed - `if ($phase == 'runtime' && backdrop_is_apache()) `

Up here it's called, although it might not be necessary anyway, and is more than 200 lines away from its actual usage (line 254, 256).

Reply:
There are several more places where it's in use, but again so far away, that it could be more readable, if it got set down there in these multiple `if ($phase == 'runtime') {`

My concern here is readability. In line 606 `$site_config` comes "out of nothing"... Per se your change is correct, though, so I'll leave it to you to decide.

Reply:
I had already started to make the recommended changes, when I noticed there are multiple instances of `if ($phase == 'runtime') {` condition. Which means if we move config read inside of `if` condition, then we would have multiple reads. So let's leave the config read in the very beginning of the function and re-use $config object anywhere we need.